DOC-2039: Added TINY-9601 to the release notes. (#2750)
tinymce/tinymce-docs: modules/ROOT/pages/6.5-release-notes.adoc

== Additions
{productname} 6.5 also includes the following additions:

=== New `DataTransfer`, `DataTransferContent`, `DataTransferEvent`, and `DataTransferMode` APIs to handle creating and manipulating `DataTransfer` objects.
//#TINY-9601

> Dragging and dropping `contenteditable="false"` elements within {productname} emits dragstart, drop, and dragend events as expected. However, in previous versions of {productname}, these drag events had their `dataTransfer` property set to null, whereas according to web standards it should be set to a `DataTransfer` object holding the drag operation's data.

Reviewer: ```suggestion
Dragging and dropping `contenteditable="false"` elements within {productname} emits `dragstart`, `drop`, and `dragend` events as expected. However, in previous versions of {productname}, these drag events had their `dataTransfer` property set to null, whereas according to https://developer.mozilla.org/en-US/docs/Web/API/DragEvent/dataTransfer[web standards] it should be set to a `DataTransfer` object holding the drag operation's data.
```

As a consequence, integrators were unable to view and manipulate the HTML drag data in these drag events.

> In {productname} 6.5, dragstart, drop, and dragend events now have their `dataTransfer` property populated with a `DataTransfer` object whose text/html data holds the outerHTML of the `contenteditable="false"` element being dragged. Additionally, a custom mode module has been implemented to ensure that the `DataTransfer` object is in read/write mode in dragstart, read-only mode in drop, and protected mode in dragend, as per standard.

Reviewer: ```suggestion
In {productname} 6.5, dragstart, drop, and dragend events now have their `dataTransfer` property populated with a `DataTransfer` object whose `text/html` data holds the outerHTML of the `contenteditable="false"` element being dragged. Additionally, a custom mode module has been implemented to ensure that the `DataTransfer` object is in read-write mode in `dragstart` events, read-only mode in `drop` events, and protected mode in `dragend` events, as per https://html.spec.whatwg.org/dev/dnd.html#concept-dnd-rw[standard].
```

> === Exposed `dataTransfer` property of drag and drop events for elements with a `contenteditable="false"` attribute.
> //#TINY-9601
> 
> Dragging and dropping `contenteditable="false"` elements within {productname} emits `dragstart`, `drop`, and `dragend` events as expected. However, in previous versions of {productname}, these drag events had their `dataTransfer` property set to null, whereas according to https://developer.mozilla.org/en-US/docs/Web/API/DragEvent/dataTransfer[web standards] it should be set to a `DataTransfer` object holding the drag operation's data.
> 
> As a consequence, integrators were unable to view and manipulate the HTML drag data in these drag events.
> 
> In {productname} 6.5, dragstart, drop, and dragend events now have their `dataTransfer` property populated with a `DataTransfer` object whose `text/html` data holds the outerHTML of the `contenteditable="false"` element being dragged. Additionally, a custom mode module has been implemented to ensure that the `DataTransfer` object is in read-write mode in `dragstart` events, read-only mode in `drop` events, and protected mode in `dragend` events, as per https://html.spec.whatwg.org/dev/dnd.html#concept-dnd-rw[standard].
> 
> [NOTE]
> Integrators who wish to manipulate the HTML drag data should do so in the dragstart event using the `DataTransfer.setData` method.

Reviewer 2: ```suggestion
=== Exposed `dataTransfer` property of drag and drop events for elements with a `contenteditable="false"` attribute.
//#TINY-9601

Dragging and dropping elements with a `contenteditable="false"` attribute within {productname} emits `dragstart`, `drop`, and `dragend` events as expected.

However, in previous versions of {productname}, these drag events had their `dataTransfer` property set to null. According to https://developer.mozilla.org/en-US/docs/Web/API/DragEvent/dataTransfer[web standards] it should be set to a `DataTransfer` object holding the drag operation’s data.

As a consequence, integrators were unable to view and manipulate the HTML drag data in these drag events.

In {productname} 6.5, dragstart, drop, and dragend events now have their `dataTransfer` property populated with a `DataTransfer` object whose `text/html` data holds the outerHTML of the `contenteditable="false"` element being dragged.

Additionally, a custom mode module has been implemented to ensure that the `DataTransfer` object is in read-write mode in `dragstart` events, read-only mode in `drop` events, and protected mode in `dragend` events, as per https://html.spec.whatwg.org/dev/dnd.html#concept-dnd-rw[standard].

NOTE: Integrators who wish to manipulate the HTML drag data should do so in the dragstart event using the `DataTransfer.setData` method.
```
Copy-edits and some paragraphing.
